packages/mask/src/plugins/RedPacket/SNSAdaptor/hooks/useRedPacketHistory.ts

```diff
@@ -81,7 +80,8 @@ export function useRedPacketHistory(address: string, chainId: ChainId) {
                 return []
             }
         })
-
         return RedPacketRPC.getRedPacketHistoryFromDatabase(payloadList)
-    }, [address, chainId, connection])
+    }, [address, chainId])
+
+    return { ...result, value: result.value?.filter((x) => x.chainId === chainId) }
```

Comment:
Filter in above `useAsyncRetry` to avoid recreate new array during every execution.